### PR TITLE
fix(MOPLAT-490): Fix android unable to build issue

### PR DESCRIPTION
**Pull request by lordkiz**
This PR resolves [MOPLAT-490] 

### Description

- In summary: An update to RN (v0.71) caused lower versions to fetch 0.71 instead of their specified versions. Unfortunately our version still uses the gradle dynamic versioning pattern (i.e using `+` instead of specifying a specific version) so we ended up fetching v0.71. A patch for 0.66.* was released. This PR updates the version to that patch.

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Fix android unable to build issue - kizito

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-490]: https://artsyproduct.atlassian.net/browse/MOPLAT-490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: patches/react-native+0.66.5.patch [hunk 1]
```diff
@@ -56,7 +56,7 @@ index b4dab8e..2195f8e 100644
  #import "RCTSurfaceView.h"
  #import "RCTTouchEvent.h"
  #import "RCTUIManager.h"
-@@ -112,6 +113,16 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithTarget : (id)target action : (SEL)act
+@@ -112,6 +113,16 @@ - (void)_recordNewTouches:(NSSet<UITouch *> *)touches
```
Comment (brainbicycle) [inline patches/react-native+0.66.5.patch:59]: this change to the patch looks a little funky at first glance, was this just reapplying after the update?

Comment (lordkiz) [reply patches/react-native+0.66.5.patch:59]: @brainbicycle the touch related changes seems to be specific to `0.66.5`. I was initially struggling with the first version of the generated patch as it was 2500+ lines long while the original patch we had was only 158 lines long. I had to delete node_modules, pods, clean the build folder and terminate all processes before I could generate this which has all the patch from 0.66.4 + a smaller patch related to UITouch. 
I am not sure where the UITouch part is coming from, but I reckon it is specific for 0.66.5.
Can you pull this branch, delete this patch and try to generate a new one after clearing your node modules etc and see if you get the same thing?

**Comment by lordkiz** [comment]
Do we need Gradle clean and clean build folder on CI for the `test-app-android` and `test-app-ios` steps to pass?

File: patches/react-native+0.66.5.patch (hunk 1 above)
Comment (MounirDhahri) [reply patches/react-native+0.66.5.patch:59]: we probably have a description for this fix within the `Hacks.md` if so we can try out if everything is fine then maybe no action is needed

**Review by MounirDhahri — APPROVED** [review]
Thanks for fixing this Kizito 🙏

**Comment by lordkiz** [comment]
Works fine at my ends. Will merge. We can always revert if folks find issues.

**Pull request merged**